.github/workflows/update-fetchgit.yml

```diff
@@ -0,0 +1,95 @@
+name: Update fetchGit repositories
+
+on:
+  push:
+
+  pull_request:
+
+  workflow_dispatch:
+
+  schedule:
+    # run three times a day
+    - cron: "0 4,12,20 * * *"
+
+jobs:
+  update-nix-fetchgit:
+    runs-on: ubuntu-latest
+
+    permissions:
+      # for git push
+      contents: write
+      # to allow opening the PR
+      pull-requests: write
+
+    env:
+      BRANCH_NAME: "update-fetchgit"
+
+    steps:
+      - name: Install Nix
+        uses: cachix/install-nix-action@v25
+        with:
+          # needed at least for act
+          # TODO: I have no idea how to properly fill this variable... sigh...
+          #nix_path: nixpkgs=channel:nixpkgs-unstable
+          # avoids rate limits
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install update-nix-fetchgit
+        run: |
+          # TODO: should be doable with nix_path above
+          nix-channel --add https://nixos.org/channels/nixpkgs-unstable
+          nix-channel --update
+
+          nix-env -i update-nix-fetchgit
+
+      - uses: actions/checkout@v4
+        with:
+          ref: "${{ env.BRANCH_NAME }}"
+          # needed to be able to update the branch! default depth is 1! 0 means unlimited!
+          fetch-depth: 0
+
+      - name: Configure git
+        run: |
+          git config --global user.name "update-nix-fetchgit bot"
+          git config --global user.email "update-nix@fetchgit.bot"
+
+      - name: Update from master branch
+        run: |
+          if ! git pull --rebase origin master; then
+            echo "Rebase failed, merge-pulling"
+            git rebase --abort
+            git pull --no-rebase origin master
```

Comment:
I don't see any situation where explicit merges work where a rebase wouldn't. Would you mind elaborating why this would be necessary? Perhaps I'm missing something.

Reply:
This is an older line of code written while I hadn't set the checkout depth correctly (default is 1). Using the merge workflow didn't do any good there either.

There are so many situations to pick from. For instance, when a PR is merged by hand or squashed improperly or when branches are based on one another... we could make it write a comment instead, but then again you'll have a comment for every run... well at least it'd ensure people's attention.